src/prototype_rust_sdk/workflow_future.rs

```diff
@@ -303,16 +352,20 @@ impl Future for WorkflowFuture {
                         );
                     }
                     RustWfCmd::SubscribeSignal(signame, chan) => {
-                        self.sig_chans.insert(signame, chan);
+                        // Deal with any buffered signal inputs for signals that were not yet
+                        // registered
+                        if let Some(SigChanOrBuffer::Buffer(buf)) = self.sig_chans.remove(&signame)
+                        {
+                            for input in buf {
+                                let _ = chan.send(input);
+                            }
+                            // Re-poll wf future since signals may be unblocked
+                            res = self.inner.poll_unpin(cx);
+                        }
+                        self.sig_chans.insert(signame, SigChanOrBuffer::Chan(chan));
```

Comment:
It can't be done at that point unfortunately. It needs to send the send half of the channel to the context where it can be stored, but that can't happen until the underlying workflow code is polled, but we don't want to do that yet. It's a similar problem to what we had in node.

My expectation for the Rust SDK is this will go away by registering signals along with the workflow code statically, by using a struct or something. Dynamically registered signals will maybe still be a reason to have this though.